core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -881,11 +881,14 @@ class Partition(val topicPartition: TopicPartition,
   private def isReplicaIsrEligible(followerReplicaId: Int): Boolean = {
     metadataCache match {
       // In KRaft mode, only replicas which are not fenced nor in controlled shutdown are
-      // allowed to join the ISR. This does not apply to ZK mode.
+      // allowed to join the ISR. In ZK mode, we just ensure the broker is alive and not shutting down.
```

Comment:
In ControllerChannelManager.sendUpdateMetadataRequests(), it seems that we include shutting down broker in liveBrokers. So, we won't know whether a remote broker is shutting down.

Reply:
Yeah -- this was something I was looking into. I'm not sure if there is a way to also exclude shutting down brokers here. Is that also included in the metadata? I can take a look as well.

Reply:
I don't think it's possible with ZK controller on the broker-side, but having the controller-side check is probably sufficient.

Reply:
Ok makes sense. Should I change the comment to reflect that this will not block shutting down brokers here, but will be blocked controller side?

I think for at least this PR (which we want to get into 3.3) we should hold off on protocol changes.

Reply:
I think our main defense is on the follower side. We prevent the `LeaderAndIsr` from starting up the fetcher if `ReplicaManager` is shutting down. Seems ok if this check is imperfect.

nit: can we move the comments down to the respective cases?

Reply:
That makes sense. We can say that we ensure that the replica is online but that it could be in controller shutdown.

Reply:
Ah my comment is slightly different in the latest commit. Let me know if I should change it